tool/tctl/common/resource_command.go

```diff
@@ -437,6 +437,9 @@ func (rc *ResourceCommand) createRole(ctx context.Context, client auth.ClientI,
 		// check for syntax errors in predicates
 		return trace.Wrap(err)
 	}
+	if err := services.CheckDynamicLabelsInDenyRules(role); err != nil {
+		return trace.Wrap(err)
+	}
```

Comment:
nit: the error message returned by `CheckDynamicLabelsInDenyRules` is a bit vague.  Since this location is the most likely location where users will be forced to confront this error directly, having a more verbose message (like you're already doing with the alert) would probably be beneficial.